legacy_code/src/s3.py

```diff
@@ -60,7 +60,7 @@ def get_all_s3_logfiles(self):
 
     def get_all_csv(self):
         get_last_modified = lambda x: int(x.last_modified.strftime('%s'))
-        files = [f for f in self.hot_bucket.objects.all() if self.csv_check(f.key)]
+        files = [f for f in self.hot_bucket.objects.all().limit(1000) if self.csv_check(f.key)]
```

Comment:
before you implied that the limits were what was causing the timeouts- does removing this limit cause other issues.

Also, if we are actively using this code, I don't think we should call it legacy code.

Reply:
1. yes, suppose each invoke can upload 300 files, and we have 100K files in the bucket. In this situation, this API call may takes 85% execution time, and can only handle 15% * 300 = 45 files in this invoke. It is wasting time.

2. sure, we can give it another name